foundation/foundation-configuration-store/src/main/java/com/boozallen/aissemble/configuration/mutating/webhook/ConfigMutatingWebhook.java

```diff
@@ -33,31 +48,124 @@
-                                                    .withAllowed(true)
-                                                    .withUid(request.getUid())
-                                                    .build();
+        JsonFactory factory = JsonFactory.builder()
+                .enable(JsonReadFeature.ALLOW_BACKSLASH_ESCAPING_ANY_CHARACTER)
+                .enable(JsonReadFeature.ALLOW_UNESCAPED_CONTROL_CHARS)
+                .enable(JsonReadFeature.ALLOW_JAVA_COMMENTS)
+                .enable(JsonReadFeature.ALLOW_YAML_COMMENTS)
+                .enable(JsonReadFeature.ALLOW_SINGLE_QUOTES)
+                .build();
+
+        ObjectMapper mapper = new ObjectMapper(factory);
+        AdmissionResponse response;
+        try {
+            JsonNode requestJson = mapper.readTree(new StringReader(mapper.writeValueAsString(request)));
+            JsonNode source = requestJson.get("object");
+
+            StringBuilder target = replaceConfigStoreKeyWithConfigValue(source.toString());
+
+
+            if (target.length() > 1) {
+                JsonNode patch = JsonDiff.asJson(source, mapper.readTree(target.toString()));
+                response = new AdmissionResponseBuilder()
+                        .withAllowed(true)
+                        .withUid(request.getUid())
+                        .withPatch(Base64.getEncoder().encodeToString(patch.toString().getBytes()))
+                        .withPatchType("JSONPatch")
+                        .build();
+                logger.info("The kubernetes resource request has been updated.");
+            } else {
+                response = new AdmissionResponseBuilder()
+                        .withAllowed(true)
+                        .withUid(request.getUid())
+                        .build();
+                logger.info("No updates have made to the kubernetes resource request.");
+            }
+
+        } catch (Exception e) {
+            logger.error("Error injecting the config store value", e);
+            // create failed response
+            response = new AdmissionResponseBuilder()
+                    .withAllowed(false)
+                    .withUid(request.getUid())
+                    .withStatus(new StatusBuilder().withStatus("Failed").build())
+                    .build();
+        }
+
+        return new AdmissionReviewBuilder()
+                .withResponse(response)
+                .withApiVersion(admissionReviewRequest.getApiVersion())
+                .build();
+    }
+
+    public String getProperty(String groupName,String propertyName) {
+        PropertyKey propertyKey = new PropertyKey(groupName, propertyName);
+        Property property;
+        try {
+            property = this.configLoader.read(propertyKey);
+            return property.getValue();
+        } catch (Exception e) {
+            logger.error(String.format("Error reading property - groupName: %s, propertyName: %s", groupName, propertyName), e);
```

Comment:
S: We should just re-throw an exception here and allow it to bubble up to the top where we return the fail response to k8s.